README.md

```diff
@@ -1,17 +1,140 @@
+
+# 2 IPLD Format
+
+## 2.1 Container
+
+Unlike the JWT format, the IPLD encoding of UCAN does not require separate header, claims, and signature fields.
+
+```ipldsch
+type UCAN struct {
+  version String
+
+  issuer DID
+  audience DID
+  signature Signature
+
+  capabilities [Capability]
+  proofs [&UCAN]
+  expiration Int
+
+  facts [Fact]
+  nonce optional String
+  notBefore optional Int
+} representation map {
+  field facts default []
+  field proofs default []
+}
+```
+
+## 2.2 Capability
+
+``` ipldsch
+type Capability struct {
+  with Resource
+  can Ability
+  extension optional Extension
+} representation map
+```
+
+### 2.2.1 Extensions
+
+Extended capability fields contain any additional domain specific details and/or restrictions of the capability.
+
+``` ipldsch
+type Extension = { String: Any }
+```
+
+## 2.3 Fact
+
+``` ipldsch
+type Fact { String: Any }
+```
+
+## 2.4 Resource
+
+The resource pointer MUST be formatted as a [URI](https://www.rfc-editor.org/rfc/rfc3986).
+
+``` ipldsch
+type Resource = String
+```
+
+## 2.5 Ability
+
+Abilities MUST be lowercase, and MUST be namespaced with `/` delimeters. Abilities SHOULD have at least one path segment.
+
+``` ipldsch
+type Ability = String
+```
+
+## 2.6 DID
+
+DIDs MUST be encoded as [DID](https://www.w3.org/TR/did-core/)s. The [`did:key` method](https://w3c-ccg.github.io/did-method-key/) is RECOMMENDED as it is self-contained.
+
+``` ipldsch
+type DID = String
+```
+
+## 2.7 Proof
+
+As of UCAN 0.9, all proofs MUST be given as links (CIDs). Note that UCANs MAY be inlined via the [identity multihash (`0x00`)](https://github.com/multiformats/multicodec/blob/master/table.csv#L2) CID.
+
+``` ipldsch
+&UCAN
+```
+
+## 2.8 Signature
+
+The signature MUST be computed by first encoding as a [base64url JWT](#3-jwt-canonicalization), and then signed with the issuer's private key.
+
+``` ipldsch
+type Signature = Bytes
+```
+
+# 3 JWT Canonicalization
+
+Per the core UCAN spec, all implementations MUST support JWT encoding. This provides a common representation that all implementations can understand. JWT canonicalization allows for an IPLD UCAN to be expressed as a JWT, retain the JWT signature scheme, and so on for compatibilty, while retaining the ability to translate into other formats for storage or transmission among IPLD-enabled peers.
+
+To canonicalize an IPLD UCAN to JWT, the JSON segments MUST be encoded per the [JSON Canonicalization Scheme (JCS)](https://www.rfc-editor.org/rfc/rfc8785), encoded as unpadded [base64url](https://datatracker.ietf.org/doc/html/rfc4648#section-5), and joined with `.`s.
+
+## 3.1 Encoding Header
+
+When serialized to JWT, an encoding header MUST be included in the format: `enc: "jcs"`. This signals that the UCAN was intended to be encoding agnostic, and that a JWT may be reencoded. Note that while it is possible to produce an otherwise spec conformant UCAN, this MUST be treated as raw bytes, not IPLD.
```

Comment:
Probably controversial!

Reply:
(Shows as `outdated` due to text changes, but the actual expected-to-be-controversial change is still there)

Reply:
I like the `"enc":"JCS"` header, I think that may be very useful signalling.

I don't understand the last sentence though:
> Note that while it is possible to produce an otherwise spec conformant UCAN, this MUST be treated as raw bytes, not IPLD.

What does "this" refer to here? `"enc":"JCS"`?
What does "treated as raw bytes" mean?

Reply:
LOL oops I guess I must have removed the link to JCS https://www.rfc-editor.org/rfc/rfc8785

Will add it back in 👍

Reply:
Oh right. that reminds me of something else: Does `dag-json` follow the JCS spec? Or is it a thing of its own?
If not, then this might be a question of dag-json vs. JCS here.

Reply:
Also, even with that link, I don't think that clarifies my questions regarding that sentence :eyes:

Reply:
> dag-json vs. JCS here

That is a really good point, yeah. Lemme check dag-json (and dag-jose) but that really should be what determines the format 💯

Reply:
> treated as raw bytes

Yeah I'll clarify that too. The CID multicodec needs to treat a UCAN without this header as `raw` https://github.com/multiformats/multicodec/blob/master/table.csv#L39

Reply:
Aha. So `this` here refers to "a UCAN that does not have the `enc` header value set to `JCS`.

And trying to parrot to make sure I'm understanding correctly: ucan-ipld UCANs will hash to, let's say `dag-cbor`-coded CIDs, while non-ucan-ipld UCANs will hash to `raw`-coded CIDs, right?

Reply:
Oh the canonicalization was in the section above. Oh well, getting changed to dag-json anyways :P Also, I think maybe we cut the header flag because its CID will flag this for us.

Reply:
> let's say dag-cbor-coded CIDs, while non-ucan-ipld UCANs will hash to raw-coded CIDs, right?

Correct!